packages/bee-q/src/components/panel/bq-panel.tsx

```diff
@@ -0,0 +1,147 @@
+import { h, Component, Prop, Element, Method, Watch, EventEmitter, Event } from '@stencil/core';
+
+import { FloatingUIPlacement } from '../../services/interfaces';
+import { FloatingUI } from '../../services/libraries';
+
+/**
+ * @part base - The `div` element which acts as a container for the panel content
+ */
+@Component({
+  tag: 'bq-panel',
+  styleUrl: './scss/bq-panel.scss',
+  shadow: true,
+})
+export class BqPanel {
+  // Own Properties
+  // ====================
+
+  private panel: HTMLElement;
+  private floatingUI: FloatingUI;
+  private trigger: HTMLElement;
+
+  // Reference to host HTML element
+  // ===================================
+
+  @Element() el!: HTMLBqPanelElement;
+
+  // State() variables
+  // Inlined decorator, alphabetical order
+  // =======================================
+
+  // Public Property API
+  // ========================
+
+  /** Distance (px) between the panel and the trigger element. */
+  @Prop({ reflect: true }) distance?: number = 0;
+
+  /** Position of the panel */
+  @Prop({ reflect: true }) placement?: FloatingUIPlacement = 'bottom';
+
+  /** If true, panel is visible.
+   * You can toggle this attribute to show/hide the panel.
+   */
+  @Prop({ reflect: true }) open?: boolean = false;
+
+  /** If true, the scrollbar is visible.
+   * You can toggle this attribute to show/hide the scrollbar.
+   */
+  @Prop({ reflect: true }) scrollbar?: boolean = false;
+
+  // Prop lifecycle events
+  // =======================
+
+  @Watch('open')
+  handleOpenProp() {
+    this.bqPanelVisibility.emit(this.open);
+  }
+
+  @Watch('distance')
+  @Watch('placement')
+  onPropChange() {
+    this.floatingUI.init({
+      placement: this.placement,
+      distance: this.distance,
+      sameWidth: false,
+      strategy: 'fixed',
+      skidding: 0,
+    });
+  }
+
```

Comment:
When passing the trigger element through a property, we can have a `@Watch` for it:

```js
  @Watch('triggerElement')
  onTriggerElementChange() {
    if (!this.triggerElement) return;

    this.floatingUI = new FloatingUI(this.triggerElement, this.panel, { ...this.options });
  }
```

PS: `this.options` is a private get that maps the FloatingUI options